projects/plugins/jetpack/tests/e2e/bin/slack.js

```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/env node
+
+const { WebClient, ErrorCode, retryPolicies, LogLevel } = require( '@slack/web-api' );
+const fs = require( 'fs' );
+const config = require( 'config' );
+const path = require( 'path' );
+const yargs = require( 'yargs' );
+const { fileNameFormatter } = require( '../lib/utils-helper' );
+const slackClient = new WebClient( config.get( 'slack.token' ), {
+	retryConfig: retryPolicies.rapidRetryPolicy,
+	logLevel: LogLevel.ERROR,
+} );
+const slackChannel = config.get( 'slack.channel' );
+
+// region yargs
+// eslint-disable-next-line no-unused-expressions
+yargs
+	.usage( 'Usage: $0 <cmd>' )
+	.demandCommand( 1, 1 )
+	.command(
+		'run [suite]',
+		'Sends a Slack notification with test run results.',
+		() => {
+			yargs.positional( 'suite', {
+				describe: 'Test suite name',
+				type: 'string',
+			} );
+		},
+		async argv => await reportTestRunResults( argv.suite )
+	)
+	.command(
+		'job <status>',
+		'Sends a Slack notification with CI job status which can include more test runs.',
+		() => {
+			yargs.positional( 'status', {
+				describe: 'Job status',
+				type: 'string',
+			} );
+		},
+		async argv => await reportJobRun( argv.status )
+	)
+	.help( 'h' )
+	.alias( 'h', 'help' ).argv;
+
+//endregion
+
+// region main methods
+/**
+ * Sends a Slack notification with test run results.
+ * Content is built from test output (reports, logs, screenshots, etc.) and Github env variables.
+ *
+ * @param {string} suite test suite name
+ * @return {Promise<void>}
+ */
+async function reportTestRunResults( suite = 'Jetpack e2e tests' ) {
+	let result;
+
+	// If results summary file is not found send failure notification and exit
+	try {
+		result = JSON.parse( fs.readFileSync( 'output/summary.json', 'utf8' ) );
+	} catch ( error ) {
+		const errMsg = 'There was a problem parsing the test results file.';
+		console.error( errMsg );
+		await sendMessage( buildDefaultMessage( false, errMsg ), {} );
+		return;
+	}
+
+	const detailLines = [];
+	const failureDetails = [];
+	const screenshots = fs.readdirSync( config.get( 'dirs.screenshots' ) );
+	const matchedScreenshots = [];
+
+	// Go through all test results and extract failure details
+	for ( const tr of result.testResults ) {
+		for ( const ar of tr.assertionResults ) {
+			if ( ar.status !== 'passed' ) {
```

Comment:
sending log files to the slack could be useful when looking through failures in multiple runs

Reply:
Done